src/sql/src/plan/statement/ddl.rs

```diff
@@ -602,13 +602,12 @@ pub fn plan_create_source(
             key: _,
         }) => {
             let connection_item = scx.get_item_by_resolved_name(connection_name)?;
-            let mut kafka_connection = match connection_item.connection()? {
-                Connection::Kafka(connection) => connection.clone(),
-                _ => sql_bail!(
+            if !matches!(connection_item.connection()?, Connection::Kafka(_)) {
+                sql_bail!(
```

Comment:
Uncovered, I tried adding a test for this:
```diff
diff --git a/test/sqllogictest/name_resolution.slt b/test/sqllogictest/name_resolution.slt
index a5fd64ba3b..08294d47f3 100644
--- a/test/sqllogictest/name_resolution.slt
+++ b/test/sqllogictest/name_resolution.slt
@@ -64,3 +64,9 @@ CREATE CONNECTION pg TO POSTGRES (
   DATABASE postgres,
   USER postgres
 ) WITH (VALIDATE = false)
+
+statement ok
+CREATE TABLE not_a_connection (a int)
+
+statement error not_a_connection is not a kafka connection
+CREATE SOURCE pg_source FROM KAFKA CONNECTION not_a_connection (TOPIC 'topic') FORMAT TEXT ENVELOPE NONE
```
But get a different error, with unresolved db/schema name btw:
```
PlanFailure:test/sqllogictest/name_resolution.slt:72:
db error: ERROR: connection "u2.u6.not_a_connection" does not exist: ERROR: connection "u2.u6.not_a_connection" does not exist
```
Am I misunderstanding how to trigger this part of the code?

Reply:
Yeah, this checks that the connection is a Kafka connection; if it isn't a connection you get the error you saw. Pushed up a commit that tests this. However, if you use a non-Kafka connection you get an error during purification (as opposed to in planning, which is where this code lives). To trigger this error you would have to have created a catalog in a bad state--i.e. this error is unreachable/purely defensive.